src/cpp/src/lora_adapter.cpp

```diff
@@ -490,13 +494,47 @@ class LoRATransformBase : public ov::pass::MatcherPass {
 
 };
 
+// Replaces alpha with scale using LoRA rank, if the rank could be computed -
+// to align with PEFT:self.scaling[active_adapter] = self.lora_alpha[active_adapter] / self.r[active_adapter]
+std::shared_ptr<ov::Node> scale_alpha(
+    const std::shared_ptr<ov::Node>& A,
+    const std::shared_ptr<ov::Node>& B,
+    const std::shared_ptr<ov::Node>& alpha,
+    const ov::element::Type& element_type)
+{
+    using namespace ov::op;
+
+    // rank_A = A_shape[0] and rank_B = B_shape[1]
+    auto axis_0 = v0::Constant::create(ov::element::i32, {}, {0});
+    auto rank_A = std::make_shared<v8::Gather>(std::make_shared<v3::ShapeOf>(A), v0::Constant::create(ov::element::i32, {1}, {0}), axis_0);
+    auto rank_B = std::make_shared<v8::Gather>(std::make_shared<v3::ShapeOf>(B), v0::Constant::create(ov::element::i32, {1}, {1}), axis_0);
+
+    // scaled_alpha = alpha / rank
+    auto lora_rank = std::make_shared<v0::Convert>(rank_A, element_type);
+    auto scaled_alpha = std::make_shared<v1::Divide>(alpha, lora_rank);
+
+    // result = ranks_equal ? scaled_alpha : alpha
+    auto ranks_equal = std::make_shared<v1::Equal>(rank_A, rank_B);
+    return std::make_shared<v1::Select>(ranks_equal, scaled_alpha, alpha);
```

Comment:
Minor, not sure that `select` perf is ok

Reply:
I'm not sure how effective this approach is overall, but let's keep this PR as a backup plan